core/src/main/kotlin/fr/sncf/osrd/api/api_v2/path_properties/PathPropRequest.kt

```diff
@@ -0,0 +1,25 @@
+package fr.sncf.osrd.api.api_v2.path_properties
+
+import com.squareup.moshi.Json
+import com.squareup.moshi.JsonAdapter
+import com.squareup.moshi.Moshi
+import com.squareup.moshi.kotlin.reflect.KotlinJsonAdapterFactory
+import fr.sncf.osrd.api.api_v2.pathfinding.TrackRange
+
+class PathPropRequest(
+    var trackRanges: List<TrackRange>,
+    @Json(name = "rolling_stock_supported_electrifications")
+    var rsSupportedElectrifications: List<String>,
+    var infra: String,
+    /** The expected infrastructure version */
+    @Json(name = "expected_version") var expectedVersion: String,
+) {
+
+    companion object {
+        val adapter: JsonAdapter<PathPropRequest> =
+            Moshi.Builder()
+                .add(KotlinJsonAdapterFactory())
+                .build()
+                .adapter(PathPropRequest::class.java)
+    }
```

Comment:
You need the distance adapters (see [here](https://github.com/OpenRailAssociation/osrd/blob/dev/core/src/main/kotlin/fr/sncf/osrd/api/api_v2/pathfinding/PathfindingBlockResponse.kt#L98-L105))